### PR TITLE
Fix dtype inference for ``engine="pyarrow"`` in ``read_csv``

### PR DESCRIPTION
**Pull request by phofl**
- [x] Tests added / passed
- [x] Passes `pre-commit run --all-files`

this is raising right now

### CODE REVIEW

File: dask/dataframe/io/csv.py [hunk 1]
```diff
@@ -620,6 +620,9 @@ def read_pandas(
     # Use sample to infer dtypes and check for presence of include_path_column
     head_kwargs = kwargs.copy()
     head_kwargs.pop("skipfooter", None)
+    if head_kwargs.get("engine") == "pyarrow":
+        # Use c engine to infer since Arrow engine does not support nrows
+        head_kwargs["engine"] = "c"
```
Comment (jrbourbeau) [inline dask/dataframe/io/csv.py:625]: Should we set this to `None` instead to use whatever the default in `pandas` is? Probably not a huge deal either way, just wondering if there's any downside to hardcoding `"c"`

**Review by jrbourbeau — COMMENTED** [review]
Nice -- thanks @phofl. Just one small question, but otherwise this looks good

File: dask/dataframe/io/csv.py (hunk 1 above)
Comment (phofl) [reply dask/dataframe/io/csv.py:625]: We could pop the key, None wouldn't work. I don't think that we would ever change it in pandas, but better be save

Comment (phofl) [reply dask/dataframe/io/csv.py:625]: Changed my mind, we might switch to arrow at some point, better to keep c here

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**